src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -2016,6 +2110,36 @@ impl<'a> QemuCmdLine<'a> {
         self.devices.push(Box::new(balloon_device));
     }
 
+    pub fn add_sev_protection_device(&mut self, cbitpos: u32, firmware: &str) {
```

Comment:
What's about just using the method `add_sev_snp_protection_device` to handle the two cases: SEV and SEV-SNP. As I notice that they have so much similar handling logic.

Reply:
I originally literally had a `add_sev_snp_protection_device()` function but I split it into two since 1) it had more parameters, not all of them always used, and 2) it had more branching in its impementation.  In the end, I thought it was harder to read than two tailor-made functions and a single point of branching (in the caller).

Admittedly the above might be a matter of taste, but if SEV is about to be dropped as mentioned by another reviewer, having two functions will make that arguably easier I believe.